subsys/mpsl/fem/nrf21540_gpio_spi/mpsl_fem_nrf21540_gpio_spi.c

```diff
@@ -0,0 +1,485 @@
+		egu_channels[i] = starting_channel + i;
+	}
+
+	return 0;
+}
+#endif // defined(CONFIG_HAS_HW_NRF_DPPIC)
+
+#if IS_ENABLED(CONFIG_MPSL_FEM_NRF21540_GPIO_SPI)
+
+static void fem_nrf21540_spi_configure(mpsl_fem_nrf21540_gpio_spi_interface_config_t *cfg)
+{
+    cfg->spi_config = (mpsl_fem_spi_config_t) {
+        .p_spim = MPSL_FEM_SPI_REG,
+#if !defined(CONFIG_PINCTRL)
+
+        .mosi_pin = {
+#if DT_NODE_HAS_PROP(MPSL_FEM_SPI_BUS, mosi_pin)
+            .p_port        = MPSL_FEM_SPI_GPIO_PORT_REG(mosi_gpios),
+            .port_no       = MPSL_FEM_SPI_GPIO_PORT_NO(mosi_gpios),
+            .port_pin      = MPSL_FEM_SPI_GPIO_PIN_NO(mosi_gpios),
+#else
+            .port_pin      = MPSL_FEM_GPIO_INVALID_PIN
+#endif
+        },
+
+        .miso_pin = {
+#if DT_NODE_HAS_PROP(MPSL_FEM_SPI_BUS, miso_pin)
+            .p_port        = MPSL_FEM_SPI_GPIO_PORT_REG(miso_gpios),
+            .port_no       = MPSL_FEM_SPI_GPIO_PORT_NO(miso_gpios),
+            .port_pin      = MPSL_FEM_SPI_GPIO_PIN_NO(miso_gpios),
+#else
+            .port_pin      = MPSL_FEM_GPIO_INVALID_PIN
+#endif
+        },
+
+        .sck_pin = {
+#if DT_NODE_HAS_PROP(MPSL_FEM_SPI_BUS, sck_pin)
+            .p_port        = MPSL_FEM_SPI_GPIO_PORT_REG(sck_gpios),
+            .port_no       = MPSL_FEM_SPI_GPIO_PORT_NO(sck_gpios),
+            .port_pin      = MPSL_FEM_SPI_GPIO_PIN_NO(sck_gpios),
+#else
+            .port_pin      = MPSL_FEM_GPIO_INVALID_PIN
+#endif
+        },
+
+#endif // !defined(CONFIG_PINCTRL)
+
+        .cs_pin = {
+#if DT_NODE_HAS_PROP(MPSL_FEM_SPI_BUS, cs_gpios)
+            .p_port        = MPSL_FEM_SPI_GPIO_PORT_REG(cs_gpios),
+            .port_no       = MPSL_FEM_SPI_GPIO_PORT_NO(cs_gpios),
+            .port_pin      = MPSL_FEM_SPI_GPIO_PIN_NO(cs_gpios),
+#else
+            .port_pin      = MPSL_FEM_GPIO_INVALID_PIN
+#endif
+        }
+    };
+
+#if defined(CONFIG_PINCTRL)
+
+#define FEM_SPI_PIN_NUM(node_id, prop, idx) \
+	NRF_GET_PIN(DT_PROP_BY_IDX(node_id, prop, idx)),
+
+#define FEM_SPI_PIN_FUNC(node_id, prop, idx) \
+	NRF_GET_FUN(DT_PROP_BY_IDX(node_id, prop, idx)),
+
+    uint8_t fem_spi_pin_nums[] = {
+		DT_FOREACH_CHILD_VARGS(
+			DT_PINCTRL_BY_NAME(DT_BUS(MPSL_FEM_SPI_IF), default, 0),
+			DT_FOREACH_PROP_ELEM, psels, FEM_SPI_PIN_NUM
+		)
+	};
+
+    uint8_t fem_spi_pin_funcs[] = {
+		DT_FOREACH_CHILD_VARGS(
+			DT_PINCTRL_BY_NAME(DT_BUS(MPSL_FEM_SPI_IF), default, 0),
+			DT_FOREACH_PROP_ELEM, psels, FEM_SPI_PIN_FUNC
+		)
+	};
+
+	for (size_t i = 0U; i < ARRAY_SIZE(fem_spi_pin_nums); i++) {
+		switch (fem_spi_pin_funcs[i]) {
+			case NRF_FUN_SPIM_SCK:
+                mpsl_fem_extended_pin_to_mpsl_fem_pin(fem_spi_pin_nums[i], &cfg->spi_config.sck_pin);
+				break;
+			case NRF_FUN_SPIM_MOSI:
+                mpsl_fem_extended_pin_to_mpsl_fem_pin(fem_spi_pin_nums[i], &cfg->spi_config.mosi_pin);
+				break;
+			case NRF_FUN_SPIM_MISO:
+                mpsl_fem_extended_pin_to_mpsl_fem_pin(fem_spi_pin_nums[i], &cfg->spi_config.miso_pin);
```

Comment:
Could run-time `mpsl_fem_extended_pin_to_mpsl_fem_pin` be replaced by some clever macro that would provide compile-time initializator for given `mpsl_fem_pin_t`

Such macro could be the placed in mpsl_fem_utils.h

Reply:
ok